functions/Export-DbaRegServer.ps1

```diff
@@ -73,47 +82,96 @@ function Export-DbaRegServer {
         [object[]]$InputObject,
         [string]$Path = (Get-DbatoolsConfigValue -FullName 'Path.DbatoolsExport'),
         [Alias("OutFile", "FileName")]
-        [string]$FilePath,
+        [ValidateScript( {
+                if ($_.FullName -notmatch "\.xml$|\.regsrvr$") {
+                    throw "The FilePath specified must end with either .xml or .regsrvr"
```

Comment:
Please perform validation within the `begin` block and use `Stop-Function` in place of `throw` 🙇